packages/teams-js/src/internal/utils.ts

```diff
@@ -416,9 +416,57 @@ export function validateId(id: string, errorToThrow?: Error): void {
   }
 }
 
-function hasScriptTags(id: string): boolean {
+export function validateUrl(url: URL, errorToThrow?: Error): void {
+  const urlString = url.toString().toLocaleLowerCase();
+  if (hasScriptTags(urlString)) {
+    throw errorToThrow || new Error('Invalid Url');
+  }
+  if (urlString.length > 2048) {
+    throw errorToThrow || new Error('Url exceeds the maximum size of 2048 characters');
+  }
+  if (!isValidHttpsURL(url)) {
+    throw errorToThrow || new Error('Url should be a valid https url');
+  }
+}
+
+/**
+ * The hasScriptTags function first decodes any HTML entities in the input string using the decodeHTMLEntities function.
+ * It then tries to decode the result as a URI component. If the URI decoding fails (which would throw an error), it assumes that the input was not encoded and uses the original input.
+ * Next, it defines a regular expression scriptRegex that matches any string that starts with <script (followed by any characters), then has any characters (including newlines),
+ * and ends with </script> (preceded by any characters).
+ * Finally, it uses the test method to check if the decoded input matches this regular expression. The function returns true if a match is found and false otherwise.
+ * @param input URL converted to string to pattern match
+ * @returns true if the input string contains a script tag, false otherwise
+ */
+function hasScriptTags(input: string): boolean {
+  let decodedInput;
+  try {
+    const decodedHTMLInput = decodeHTMLEntities(input);
+    decodedInput = decodeURIComponent(decodedHTMLInput);
+  } catch (e) {
+    // input was not encoded, use it as is
+    decodedInput = input;
+  }
   const scriptRegex = /<script[^>]*>[\s\S]*?<\/script[^>]*>/gi;
-  return scriptRegex.test(id);
+  return scriptRegex.test(decodedInput);
+}
+
+/**
+ * The decodeHTMLEntities function replaces HTML entities in the input string with their corresponding characters.
+ */
+function decodeHTMLEntities(input: string): string {
+  const entityMap = new Map<string, string>([
+    ['&lt;', '<'],
+    ['&gt;', '>'],
+    ['&amp;', '&'],
+    ['&quot;', '"'],
+    ['&#39;', "'"],
+    ['&#x2F;', '/'],
+  ]);
+  entityMap.forEach((value, key) => {
+    input = input.replace(new RegExp(key, 'gi'), value);
+  });
```

Comment:
The latest version of the PR still appears to use RegExp... but the resolved comment suggests you planned to replace it with string.replace. Is that still your plan @ndangudubiyyam ?